### PR TITLE
vscode-extensions.python-language-server: init

### PR DESCRIPTION
**Pull request by teto**
It was already defined in nix but not referencable.
I make it accessible so that one can reuse it for instance in coc-python (which otherwise installs its version in $HOME/.config/coc/extensions/coc-python-data/languageServer.0.4.24/)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by eadwu — APPROVED** [review]

File: pkgs/misc/vscode-extensions/python/language-server.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ fetchurl, extractNuGet, arch }:
+let
+
+  languageServerSha256 = {
+    linux-x64 = "1w3y0sn6ijk1vspi4lailg1q1iy9lwslhx92c7jbrrkiaszvaqwn";
+    osx-x64 = "069qwwl08fa24rsnln4i5x9mfriqaw920l6v2j8d1r0zylxnyjsa";
```
Comment (teto) [inline pkgs/misc/vscode-extensions/python/language-server.nix]: the mac sha should be wrong, let's hope CI can give the correct one,

Comment (teto) [reply pkgs/misc/vscode-extensions/python/language-server.nix]: solved

**Comment by teto** [comment]
seems like I need to patchelf the interpreter. I am trying to add hooks to extra-nuget.nix ...

**Comment by teto** [comment]
I am not sure what I am doing: I've added `autoPatchelfHook` and then added libraries until the build succeeded.
When coc-python starts the mpls server, it now errors with:
```
Couldn't find a valid ICU package installed on the system. Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode.GetGlobalizationInvariantMode()
   at System.Globalization.GlobalizationMode..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.FormattableString.Invariant(System.FormattableString)
   at Microsoft.Python.Core.Services.ServiceManager.AddService(System.Object, System.Type)
   at Microsoft.Python.LanguageServer.Services.CoreShell..ctor()
   at Microsoft.Python.LanguageServer.Services.CoreShell.Create()
   at Microsoft.Python.LanguageServer.Server.Program.Main(System.String[])
[Error  - 7:34:50 PM] Connection to server got closed. Server will not be restarted.
[Error  - 7:34:50 PM]
```
This can be prevented by writing/patching Microsoft.Python.LanguageServer.runtimeconfig.json with
```
{
"runtimeOptions": {
    "configProperties": {
        "System.Globalization.Invariant": true
    }
}
}
```
(or by adding `icu` to the buildInputs)

The follow up problem I meet is the server crashing with
```
##########Linting Output - mypy##########
No usable version of libssl was found
```
Not sure how to solve that, I guess I could strace ...
It seems like wrapping MPLS in a dotnet environment could solve several of the previous problems
https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore30

Not sure how to progress from there

**Comment by Profpatsch** [comment]
ping

**Comment by teto** [comment]
@jonringer I feel like there are different ways to complete this and I am not confident enough to modify the dotnet packages I know you are already doing a lot but as a dotnet and python expert, I wonder if you could complete/help with this :) ?

**Comment by teto** [comment]
nevermind closing in favor of https://github.com/NixOS/nixpkgs/pull/70058

**Pull request closed without merging**